foursight_core/react/api/react_routes.py

```diff
@@ -108,42 +109,59 @@ def reactapi_route_user_get_or_patch_or_delete(env: str, uuid: str) -> Response:
         if the methods are different; rather you must bundle them together and
         distinguish between which method is used programmatically as we do here.
         """
+        request = app.current_request.to_dict()
         if app.current_request.method == "GET":
-            return app.core.reactapi_get_user(app.current_request.to_dict(), env, uuid=uuid)
+            args = get_request_args(request)
+            return app.core.reactapi_get_user(request, env, uuid, args)
         elif app.current_request.method == "PATCH":
             user = get_request_body(app.current_request)
-            return app.core.reactapi_patch_user(app.current_request.to_dict(), env, uuid=uuid, user=user)
+            return app.core.reactapi_patch_user(request, env, uuid=uuid, user=user)
         elif app.current_request.method == "DELETE":
-            return app.core.reactapi_delete_user(app.current_request.to_dict(), env, uuid=uuid)
+            return app.core.reactapi_delete_user(request, env, uuid=uuid)
         else:
             return app.core.create_forbidden_response()
 
+    @route("/{env}/users/institutions", authorize=True)
+    def reactapi_route_users_projects(env: str) -> Response:  # noqa: implicit @staticmethod via @route
+        request = app.current_request.to_dict()
+        args = get_request_args(request)
+        return app.core.reactapi_users_institutions(request, env, args)
+
+    @route("/{env}/users/projects", authorize=True)
+    def reactapi_route_users_projects(env: str) -> Response:  # noqa: implicit @staticmethod via @route
+        request = app.current_request.to_dict()
```

Comment:
Why do some of these methods take args and some don't?  I don't see any doc strings that would help me understand which do and don't. Shouldn't the ones that don't take args at least complain if they get args? Shouldn't the ones that take args have to say which args they take and complain if there are others besides those?

Reply:
I feel like this dance to get the args should be done in a decorator.

Reply:
Well some endpoints take args and some don't. But I definitely should have included comments/docstrings on this, which I've added. In all cases the arguments are optional. I'm all for moving things if possible to the decorator (which I'd previously created - react_route_decorator.py) but not quite sure how the args might be moved there; lemme know if you had something in mind.

Reply:
I can probably sketch something for you out-of-band.